src/rooms/RestartRoom.ts

```diff
@@ -11,11 +11,36 @@ export class RestartRoom extends Room<RestartRoomState> {
   maxClients = 8;
   LOBBY_CHANNEL = "$restartlobby";
 
+  //Constructor for disposeTimer
+  private disposeTimer: NodeJS.Timeout | null = null;
+
+  constructor() {
+    super();
+    this.startDisposeTimer = this.startDisposeTimer.bind(this);
+  }
+
+  private startDisposeTimer() {
+    this.disposeTimer = setTimeout(() => {
+      if (this.clients.length <= 1) {
+        this.onDispose(); // Disconnect clients and dispose if conditions are met
+      }
+    }, 10 * 1000); // 10 seconds in milliseconds
+  }
+
   async onCreate(options: any) {
     this.roomId = await generateRoomId(this.presence, this.LOBBY_CHANNEL);
     console.log("Room created with ID: ", this.roomId);
     // Create logic to dispose rooms after a certain time if there is <= 1 connected client
     this.autoDispose = false;
+    this.setState(new RestartRoomState());
+
+    this.onMessage("*", (client, type, message) => {
+      messageBroker(this, client, type as string, message);
+    });
+
+    initAIEngine(this);
+
+    setTimeout(this.startDisposeTimer, 10 * 1000); // Start after 1 minute in milliseconds
```

Comment:
Use Room#clock instead of javascript setTimeout and setInterval.
Ref: https://docs.colyseus.io/server/timing-events/#server-api-timing-events